packages/react-instantsearch-core/src/lib/useInstantSearchApi.ts

```diff
@@ -106,6 +107,8 @@ export function useInstantSearchApi<TUiState extends UiState, TRouteState>(
       search._initialResults = initialResults || {};
     }
 
+    hydrateSearchClient(props.searchClient, initialResults);
```

Comment:
personally I would see this as part of the _initialResults code in InstantSearch, not in React InstantSearch